portfolio/src/main/webapp/script.js

```diff
@@ -56,6 +56,18 @@ function loadSkills () {
  */
  async function fetchFromData() {
   const response = await fetch('/data');
-  const quote = await response.text();
-  document.getElementById('secret-message').innerText = quote; 
+  // This gives a list of messages
+  const msgs = await response.json();
+
+  // Clear the child elements from the div to avoid duplicating
+  const root = document.getElementById("all-messages");
+  root.innterHTML = ""; 
```

Comment:
This might have issues for old browsers (IIRC, not entirely sure), and innerHTML is generally not preferred.

(I think using jquery to clear the dom is fine)
https://github.com/jquery/jquery/blob/90fed4b453a5becdb7f173d9e3c1492390a1441f/src/manipulation.js#L340
You can see jquery does not use innerHtml, it iterates the inner node and clear the text content.